### ISSUE TITLE
SQLiteError is impossible to catch

### ISSUE THREAD

**Issue by cmd410** [codehz/easy_sqlite3#3]
Consider the following:
```nim
import easy_sqlite3

proc createTable() {.importdb: "CREATE TABLE data(id INTEGER PRIMARY KEY, value INTEGER)".}
proc insertData(id, value: int) {.importdb: "INSERT INTO data(id, value) VALUES ($id, $value)".}

var db = initDatabase(":memory:")
db.createTable()

db.insertData(1, 1)

try:
  db.insertData(1, 2)
except:
  echo getCurrentExceptionMsg()

db.insertData(2, 4)
```
This code should run normally, but it fails with `SQLiteError` due to UNIQUE constraint violation on primary key. Interestingly enough compiler points to the last line, where there is no error instead of `db.insertData(1, 2)`. The error seems to be impossible to catch.

Truthfully I have no idea how this happens.

```console
$ nim --version
Nim Compiler Version 1.6.8 [Linux: amd64]
Compiled at 2022-09-27
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: c9f46ca8c9eeca8b5f68591b1abe14b962f80a4c
active boot switches: -d:release
```

**Comment by codehz** [comment]
Because it only occurs when the reset function is executed, but reset will be performed at the beginning of the statement execution. I'll try moving reset to the end

**Comment by codehz** [comment]
should fixes in [c55c254](https://github.com/codehz/easy_sqlite3/commit/c55c254091d285aaa0ae36d6f2c75c3ee39d1d4a)
